src/inventory.cpp

```diff
@@ -271,6 +271,30 @@ std::string ItemStack::getShortDescription(const IItemDefManager *itemdef) const
 	return desc;
 }
 
+f32 ItemStack::getRange(const IItemDefManager *itemdef) const
+{
+    std::string sRange = metadata.getString("range");
+    f32 range = -1;
+    if (sRange.empty())
+        range = getDefinition(itemdef).range;
+    if (!sRange.empty())
+    {
+        try {
+            //This is the best way (until otherwise said) to convert a string into a float.
+            //std::stof gives a compilation error along the lines of: namespace 'std' does not contain 'mystof'.
+			char *c = const_cast<char *>(sRange.c_str());
```

Comment:
I don't see why `c` is not a `const char *`, and I don't see why the `const_cast` would be necessary:

* `c_str` has the signature [`const char* c_str() const;`](https://cplusplus.com/reference/string/string/c_str/)
* `strtof` has the signature [`float strtof( const char* str, char** str_end );`](https://en.cppreference.com/w/cpp/string/byte/strtof)